projects/storefrontlib/src/shared/components/modal/modal.service.ts

```diff
@@ -0,0 +1,18 @@
+import { Injectable } from '@angular/core';
+import { NgbModal } from '@ng-bootstrap/ng-bootstrap';
+import { ModalRef } from './modal-ref';
+import { ModalOptions } from './modal-options';
+
+/**
+ * A service to open modal
+ */
+@Injectable({
+  providedIn: 'root',
+})
+export class ModalService {
+  constructor(private ngbModalService: NgbModal) {}
+
+  open(content: any, options?: ModalOptions): ModalRef {
+    return this.ngbModalService.open(content, options ? options : null);
+  }
```

Comment:
Shouldn't we provide also `close()` method? In my opinion would be useful to make it too in our facade for the future extensibility.

Reply:
currently we don't use `close()` method in our app. it's better for us to add new features in the future than changing them :) will leave as is